### PR TITLE
Feat/savings

### PR DESCRIPTION
**Pull request by vangeliq**
implemented a basic savings feature. it's basically the transaction page, except:

- it has no transaction type or payment type
- no bulk upload implemented

image

image
image

### CODE REVIEW

File: client/src/components/Savings/AddSavingItem.js [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+            <Dialog open={open} onClose={resetFields} maxWidth="md">
+                <DialogTitle>Add Saving</DialogTitle>
+                <DialogContent >
+                    <DialogContentText>
+                        Please enter the saving details below:
+                    </DialogContentText>
+                    <TextField
+                        margin="dense"
+                        id="merchantName"
+                        label="Merchant Name"
```
Comment (RebeccaLiu1) [inline client/src/components/Savings/AddSavingItem.js:92]: should we maybe change the name for savings? not sure if "merchange name" makes the most sense

File: client/src/components/Savings/UpdateSavingItem.js [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+            <Dialog open={open} onClose={resetFields} maxWidth="md">
+                <DialogTitle>Update Saving</DialogTitle>
+                <DialogContent >
+                    <DialogContentText>
+                        Please enter the new saving details below:
+                    </DialogContentText>
+                    <TextField
+                        margin="dense"
+                        id="merchantName"
+                        label="Merchant Name"
```
Comment (RebeccaLiu1) [inline client/src/components/Savings/UpdateSavingItem.js:94]: same as above ^

File: client/src/components/Savings/AddSavingItem.js (hunk 1 above)
Comment (vangeliq) [reply client/src/components/Savings/AddSavingItem.js]: changed to source!

File: client/src/components/Savings/ViewSavingsItem.js [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+                            top: 13,
+                        }}
+                    >
+                        <CloseIcon />
+                    </IconButton>
+                </DialogTitle>
+
+                <DialogContent dividers>
+                    <Typography sx={DialogHeadings}>
+                        Merchant Name:
```
Comment (RebeccaLiu1) [inline client/src/components/Savings/ViewSavingsItem.js:87]: Probably should update the heading name to source as well

**Review by RebeccaLiu1 — APPROVED** [review]

File: client/src/components/Savings/ViewSavingsItem.js (hunk 3 above)
Comment (vangeliq) [reply client/src/components/Savings/ViewSavingsItem.js:87]: thanks! i missed that

**Pull request merged**